R/derive_param_confirmed_resp.R

```diff
@@ -0,0 +1,386 @@
+#' Derive Confirmed Response Parameter
+#'
+#' Derive confirmed response parameter
+#'
+#' @param dataset Input dataset
+#'
+#'   The `PARAMCD`, `ADT`, and `AVALC` variables and the variables specified by
+#'   `subject_keys` and `reference_date` are expected.
+#'
+#' @param dataset_adsl ADSL input dataset
+#'
+#'   The variables specified for `subject_keys` are expected. For each subject
+#'   of the specified dataset a new observation is added to the input dataset.
+#'
+#' @param filter_source Source filter
+#'
+#'   All observations in `dataset_source` fulfilling the specified condition are
+#'   considered for deriving the confirmed best overall response.
+#'
+#' @param source_pd Date of first progressive disease (PD)
+#'
+#'   If the parameter is specified, the observations of the input dataset for
+#'   deriving the new parameter are restricted to observations up to the
+#'   specified date. Observations at the specified date are included. For
+#'   subjects without first PD date all observations are take into account.
+#'
+#'   *Permitted Values:* a `date_source` object (see `admiral::date_source()`
+#'   for details)
+#'
+#'   *Default:* `NULL`,
+#'
+#' @param source_datasets Source dataset for the first PD date
+#'
+#'   A named list of datasets is expected. It links the `dataset_name` from
+#'   `source_pd` with an existing dataset.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```{r eval=FALSE}
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataset in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#' @param ref_confirm Minimum time period for confirmation
+#'
+#'   The assessment and the confirmatory assessment for `"CR"` and `"PR"` have
+#'   to be at least the specified number of days apart.
+#'
+#' @param max_nr_ne
+#'
+#'   The specified number of `"NE"` assessments between the assessment and the
+#'   confirmatory assessment for `"CR"` and `"PR"` response is accepted.
+#'
+#'   *Permitted Values:* a non-negative numeric scalar
+#'
+#'   *Default:* `1`
+#'
+#' @param accept_sd Accept `"SD"` for `"PR"`?
+#'
+#'   If the argument is set to `TRUE`, one `"SD"` assessment between the
+#'   assessment and the confirmatory assessment for `"PR"` response is accepted.
+#'   Otherwise, no `"SD"` assessment must occur between the two assessments.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'   analysis value (`AVAL`)
```

Comment:
If I understood correctly, this function is for formatting avalc values to create aval values right? So, do the user has an option for eg NE=9 or NE=6? Sorry little confused as why we only talking about Y,N avalc. Is this function will only give information about the subject who had confirmed response or not ( kind of objective confirmed response Y/N). for actual confirmed BOR value( CR,PR,SD etc), we do the same calculations again in some other function?

Reply:
Yes, this argument defines the values of `AVAL` for the new parameter. The default sets `AVAL = 1` for `AVALC == "Y"` and `AVAL = 0` for `AVALC == "N"`. For this function it is unlikely that something else than the default is used. I have added the argument only for consistency with the other functions. For example `derive_param_confirmed_bor()` provides the same argument but with a different function as default.

Reply:
ok got it, thanks I will check confirmed_bor as well